### ISSUE TITLE
Updating Wrapper fails with Could not resolve all files for configuration ':classpath'.

### ISSUE THREAD

**Issue by uli-f** [gradle-update/update-gradle-wrapper-action#721]
I added a workflow with this action to my private repo:
```
name: Update Gradle Wrapper

on:
  schedule:
    # run every Sunday at 13:28
    - cron: "53 17 * * 0"
  workflow_dispatch:

jobs:
  update-gradle-wrapper:
    permissions:
      # for gradle-update/update-gradle-wrapper-action
      contents: write
      # for gradle-update/update-gradle-wrapper-action
      pull-requests: write
    runs-on: ubuntu-latest

    steps:
      - name: Checkout
        uses: actions/checkout@v3

      - name: Update Gradle Wrapper
        uses: gradle-update/update-gradle-wrapper-action@v1
        with:
          repo-token: ${{ secrets.PAT_UPDATE_GRADLE_WRAPPER_ACTION }}
          labels: "type: library-update"

      - name: Validate gradle wrapper
        uses: gradle/wrapper-validation-action@v1
```

The workflow fails at the step `Update Gradle Wrapper` and this is the output that I get:
```
Updating Wrapper
  /home/runner/work/XXX/XXX/gradlew wrapper --gradle-version 8.2 --distribution-type bin --gradle-distribution-sha256-sum 38f66cd6eef217b4c35855bb11ea4e9fbc53594ccccb5fb82dfd317ef8c2c5a3
  Downloading https://services.gradle.org/distributions/gradle-8.1.1-bin.zip
  ...........10%............20%............30%............40%............50%............60%............70%............80%...........90%............100%
  
  Welcome to Gradle 8.1.1!
  
  Here are the highlights of this release:
   - Stable configuration cache
   - Experimental Kotlin DSL assignment syntax
   - Building with Java 20
  
  For more details see https://docs.gradle.org/8.1.1/release-notes.html
  
  Starting a Gradle Daemon (subsequent builds will be faster)
  
  FAILURE: Build failed with an exception.
  
  * What went wrong:
  A problem occurred configuring root project 'XXX'.
  > Could not resolve all files for configuration ':classpath'.
     > Could not resolve org.springframework.boot:spring-boot-gradle-plugin:3.1.1.
       Required by:
           project : > org.springframework.boot:org.springframework.boot.gradle.plugin:3.1.1
        > No matching variant of org.springframework.boot:spring-boot-gradle-plugin:3.1.1 was found. The consumer was configured to find a library for use during runtime, compatible with Java 11, packaged as a jar, and its dependencies declared externally, as well as attribute 'org.gradle.plugin.api-version' with value '8.1.1' but:
            - Variant 'apiElements' capability org.springframework.boot:spring-boot-gradle-plugin:3.1.1 declares a library, packaged as a jar, and its dependencies declared externally:
                - Incompatible because this component declares a component for use during compile-time, compatible with Java 17 and the consumer needed a component for use during runtime, compatible with Java 11
                - Other compatible attribute:
                    - Doesn't say anything about org.gradle.plugin.api-version (required '8.1.1')
            - Variant 'javadocElements' capability org.springframework.boot:spring-boot-gradle-plugin:3.1.1 declares a component for use during runtime, and its dependencies declared externally:
                - Incompatible because this component declares documentation and the consumer needed a library
                - Other compatible attributes:
                    - Doesn't say anything about its target Java version (required compatibility with Java 11)
                    - Doesn't say anything about its elements (required them packaged as a jar)
                    - Doesn't say anything about org.gradle.plugin.api-version (required '8.1.1')
            - Variant 'mavenOptionalApiElements' capability org.springframework.boot:spring-boot-gradle-plugin-maven-optional:3.1.1 declares a library, packaged as a jar, and its dependencies declared externally:
                - Incompatible because this component declares a component for use during compile-time, compatible with Java 17 and the consumer needed a component for use during runtime, compatible with Java 11
                - Other compatible attribute:
                    - Doesn't say anything about org.gradle.plugin.api-version (required '8.1.1')
            - Variant 'mavenOptionalRuntimeElements' capability org.springframework.boot:spring-boot-gradle-plugin-maven-optional:3.1.1 declares a library for use during runtime, packaged as a jar, and its dependencies declared externally:
                - Incompatible because this component declares a component, compatible with Java 17 and the consumer needed a component, compatible with Java 11
                - Other compatible attribute:
                    - Doesn't say anything about org.gradle.plugin.api-version (required '8.1.1')
            - Variant 'runtimeElements' capability org.springframework.boot:spring-boot-gradle-plugin:3.1.1 declares a library for use during runtime, packaged as a jar, and its dependencies declared externally:
                - Incompatible because this component declares a component, compatible with Java 17 and the consumer needed a component, compatible with Java 11
                - Other compatible attribute:
                    - Doesn't say anything about org.gradle.plugin.api-version (required '8.1.1')
            - Variant 'sourcesElements' capability org.springframework.boot:spring-boot-gradle-plugin:3.1.1 declares a component for use during runtime, and its dependencies declared externally:
                - Incompatible because this component declares documentation and the consumer needed a library
                - Other compatible attributes:
                    - Doesn't say anything about its target Java version (required compatibility with Java 11)
                    - Doesn't say anything about its elements (required them packaged as a jar)
                    - Doesn't say anything about org.gradle.plugin.api-version (required '8.1.1')
```
All I read from that error message is that something very fundamental is going wrong here.

Any help in getting this resolved is appreciated.

**Comment by huehnerlady** [comment]
We use this action for a while and we have the same problem with spring boot 3 services. the other services with spring boot 2 seems to have worked fine

**Comment by uli-f** [comment]
Thanks @huehnerlady for your comment.

As you are a contributor to this project is there anything I might try to help getting this resolved?

**Comment by huehnerlady** [comment]
> Thanks @huehnerlady for your comment.
> 
> As you are a contributor to this project is there anything I might try to help getting this resolved?

Sorry for the confusion, I just created one Pull request updating the documentation so unfortunately I am as clueless as you are 😅

**Comment by huehnerlady** [comment]
@cristiangreco can you maybe help here? Every service we upgrade now throws that exception 😅

**Comment by huehnerlady** [comment]
@cristiangreco any update on this? more and more services have now failing github actions every week....

**Comment by uli-f** [comment]
Yes, same here. I am happy to help in any way I can to get this fixed.

**Comment by XiangRongLin** [comment]
TLDR: add setup-java action with correct java version

@huehnerlady @NilsDammenhayn Our problem is that we don't explicitly specify the java version. 
It can be seen in the log `Incompatible because this component declares a component for use during compile-time, compatible with Java 17 and the consumer needed a component for use during runtime, compatible with Java 11` 

ubuntu-latest defaults to java 11 (checked by running with debug log enabled), while we need java 17 and you @uli-f probably too. So just add the setup-java action

You can reproduce the problem locally by running `./gradlew wrapper --gradle-version 8.2.1 --distribution-type all --gradle-distribution-sha256-sum 7c3ad722e9b0ce8205b91560fd6ce8296ac3eadf065672242fd73c06b8eeb6ee`.
With java 11 it produces the error like above, with 17 it works

**Comment by uli-f** [comment]
Thank you @XiangRongLin, much appreciated!

I added the following github action to my workflow (after `actions/checkout@v3` and before `gradle-update/update-gradle-wrapper-action@v1`):

```
- name: ☕ Set up Java
  uses: actions/setup-java@v3
  with:
    distribution: 'temurin'
    java-version: '17'
```

The issue described in my [original post above](https://github.com/gradle-update/update-gradle-wrapper-action/issues/721#issue-1791776062) only ever occurred if there actually was a gradle update available. At the moment, all my projects are running on the latest gradle. But it seems like the release of gradle 8.3 isn't too far away, so I will provide an update here once the github workflow ran with a gradle update being available.

**Comment by NilsDammenhayn** [comment]
I tested some workflows with @XiangRongLin's fix and can confirm that with an update available the action now creates the pull request 👌

**Comment by huehnerlady** [comment]
@uli-f how is it going? The fix from @XiangRongLin works for us and so hopefully it does for you too? Can we close this issue?

@cristiangreco maybe some more documentation about this behaviour makes sense? 🤔

**Comment by uli-f** [comment]
Thanks @huehnerlady for reaching out.

The original issue is Indeed resolved, thanks to the support from @XiangRongLin.

I do experience another problem now though, I will close this issue and open another one.
